src/legacy/core_plugins/vis_default_editor/public/components/agg_params_helper.ts

```diff
@@ -109,19 +125,28 @@ function getAggParamsToRender({ agg, editorConfig, metricAggs, state }: ParamIns
         metricAggs,
         state,
         value: agg.params[param.name],
+        schemas,
       });
     }
   });
 
   return params;
 }
 
+const filterByName = propFilter('name');
+
 function getAggTypeOptions(
   agg: IAggConfig,
   indexPattern: IndexPattern,
-  groupName: string
+  groupName: string,
+  allowedAggs: string[],
+  aggFilter: string[]
 ): ComboBoxGroupedOptions<IAggType> {
-  const aggTypeOptions = aggTypeFilters.filter((aggTypes as any)[groupName], indexPattern, agg);
+  const aggTypeOptions = aggTypeFilters
+    .filter((aggTypes as any)[groupName], indexPattern, agg, aggFilter)
+    .filter(aggType => {
+      return filterByName([aggType], allowedAggs).length !== 0;
```

Comment:
Can't we just merge both arrays, instead of adding the second filter, which do the same thing?

Reply:
No, it won't give the same results